salt/utils/process.py

```diff
@@ -532,6 +532,23 @@ def __getstate__(self):
 
     def __setup_process_logging(self):
         salt.log.setup.setup_multiprocessing_logging(self.log_queue)
+        if not hasattr(self, '_original_run'):
+            # Patch the run method at runtime because decorating the run method
+            # with a function with a similar behavior would be ignored once this
+            # class'es run method is overridden.
+            self._original_run = self.run
+            self.run = self._run
+
+    def _run(self):
+        try:
+            return self._original_run()
+        except SystemExit:
+            # These are handled by multiprocessing.Process._bootstrap()
+            raise
+        except Exception as exc:
+            log.error(
+                'An un-handled exception from the multiprocessing process '
+                '\'%s\' was caught:\n', self.name, exc_info=True)
```

Comment:
I think you should reraise after this, otherwise the exitcode of the child comes out as 0, when 1 would be more appropriate. (Yes, you'll get a second exception printed on stderr, but I wouldn't mind.)

Other than that, this looks good to me :)